components/Blog/PostComponents/Components.tsx

```diff
@@ -0,0 +1,182 @@
+import { ReactNode } from 'react';
+import Img from 'next/image';
+import Code from './Code/Code';
+import BlockQuoteAuthor from './BlockQuoteAuthor/BlockQuoteAuthor';
+import ImageCaption from './ImageCaption/ImageCaption';
+
+interface IProps {
+  children?: ReactNode;
+  href?: string;
+}
+
+interface ImageProps {
+  src: string;
+  alt?: string;
+}
+
+function copyToClip() {
+  navigator.clipboard.writeText(window?.location?.href);
+}
+
+function idLinkCreator(children: ReactNode): string {
+  return typeof children === 'string'
+    ? children
+        // Regex to remove all punctionation from the original string
+        .replaceAll(/['!"#$%&\\'()\\*+,\-\\.\\/:;<=>?@\\[\\\]\\^_`{|}~']/g, '')
+        // Convert any spaces to '-'
+        .replaceAll(' ', '-')
+        // Replace any '--' to a '-'
+        .replaceAll(/(--)/g, '-')
+        // Make it all lower case
+        .toLowerCase()
+    : '#';
```

Comment:
Remove the `#` as the fallback value if children isn't a string, use an empty string instead.

Reply:
Resolved